consumer/heartbeats_consumer.go

```diff
@@ -0,0 +1,315 @@
+	MacAddresses   []string          `json:"mac_addresses,omitempty"`
+	FQDN           string            `json:"fqdn,omitempty"`
+	BiosUUID       string            `json:"bios_uuid,omitempty"`
+	DisplayName    string            `json:"display_name,omitempty"`
+	AnsibleHost    string            `json:"ansible_host,omitempty"`
+	CustomMetadata map[string]string `json:"custom_metadata,omitempty"`
+	Reporter       string            `json:"reporter"`
+	StaleTimestamp time.Time         `json:"stale_timestamp"`
+	QueueKey       string            `json:"queue_key,omitempty"`
+}
+
+// Request taken from https://github.com/redhatinsights/insights-ingress-go/internal/validators/types.go
+type Request struct {
+	Account     string    `json:"account"`
+	Category    string    `json:"category"`
+	ContentType string    `json:"content_type"`
+	Metadata    Metadata  `json:"metadata"`
+	RequestID   string    `json:"request_id"`
+	Principal   string    `json:"principal"`
+	OrgID       string    `json:"org_id"`
+	Service     string    `json:"service"`
+	Size        int64     `json:"size"`
+	URL         string    `json:"url"`
+	ID          string    `json:"id,omitempty"`
+	B64Identity string    `json:"b64_identity"`
+	Timestamp   time.Time `json:"timestamp"`
+}
+
+// KkfMessageProcessor Processor for kafka messages used by NewKafkaConsumer
+type KkfMessageProcessor interface {
+	HandleMessage(msg *sarama.ConsumerMessage) error
+}
+
+// KfkConsumer ...
+type KfkConsumer struct {
+	configuration    broker.Configuration
+	client           sarama.ConsumerGroup
+	Storage          storage.Storage
+	MessageProcessor KkfMessageProcessor
+	ready            chan bool
+	// cancel                               context.CancelFunc
+}
+
+// TODO
+// &HearbeatMessageProcessor{Storage: storage}
+
+// NewKfkConsumer constructs a kafka consumer with a message processor
+func NewKfkConsumer(brokerCfg broker.Configuration, processor KkfMessageProcessor) (*KfkConsumer, error) {
+	saramaConfig, err := broker.SaramaConfigFromBrokerConfig(brokerCfg)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to create sarama configuration from current broker configuration")
+		return nil, err
+	}
+	log.Info().
+		Str("addresses", brokerCfg.Addresses).
+		Str("group", brokerCfg.Group).
+		Msg("New consumer group")
+
+	consumerGroup, err := sarama.NewConsumerGroup(strings.Split(brokerCfg.Addresses, ","), brokerCfg.Group, saramaConfig)
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to create consumer group")
+		return nil, err
+	}
+	log.Info().Msg("Consumer group has been created")
+
+	consumer := &KfkConsumer{
+		configuration:    brokerCfg,
+		client:           consumerGroup,
+		MessageProcessor: processor,
+		ready:            make(chan bool),
+	}
+
+	return consumer, nil
+}
+
+// Serve starts listening for messages and processing them. It blocks current thread.
+// TODO pass ctx, cancel := context.WithCancel(context.Background())
+func (consumer *KfkConsumer) Serve(ctx context.Context) {
```

Comment:
same question as above: why not reusing the [already existing implementation of KafkaConsumer](https://github.com/RedHatInsights/insights-results-aggregator/blob/8ea82653218e805ceb9398140aad17c854c976d4/consumer/kafka_consumer.go#L153)?